supervisor/dbus/udisks2/block.py

```diff
@@ -0,0 +1,244 @@
+    options: str
+    name: str | None = None
+    passphrase_path: str | None = None
+
+    @staticmethod
+    def from_dict(data: CrypttabConfigDetailsDataType) -> "CrypttabConfigDetails":
+        """Create CrypttabConfigDetails from dict."""
+        return CrypttabConfigDetails(
+            name=bytes(data["name"]).decode(),
+            device=bytes(data["device"]).decode(),
+            passphrase_path=bytes(data["passphrase-path"]).decode(),
+            passphrase_contents=bytes(data["passphrase-contents"]).decode(),
+            options=bytes(data["options"]).decode(),
+        )
+
+    def to_dict(self) -> CrypttabConfigDetailsDataType:
+        """Return dict representation."""
+        data = {
+            "name": bytearray(self.name) if self.name else None,
+            "device": bytearray(self.device),
+            "passphrase-path": bytearray(self.passphrase_path)
+            if self.passphrase_path
+            else None,
+            "passphrase-contents": bytearray(self.passphrase_contents),
+            "options": bytearray(self.options),
+        }
+        for key in ("name", "passphrase-path"):
+            if not data[key]:
+                data.pop(key)
+        return data
+
+
+class UDisks2Block(DBusInterfaceProxy):
+    """Handle D-Bus interface for UDisks2 block device object.
+
+    http://storaged.org/doc/udisks2-api/latest/gdbus-org.freedesktop.UDisks2.Block.html
+    """
+
+    def __init__(self, object_path: str) -> None:
+        """Initialize NetworkConnection object."""
+        self.object_path = object_path
+        self.properties = {}
+
+    async def connect(self):
+        """Connect to D-Bus."""
+        self.dbus = await DBus.connect(DBUS_NAME_UDISKS2, self.object_path)
+        await self.update()
+
+    @property
+    @dbus_property
+    def device(self) -> str:
+        """Return device file."""
+        return bytes(self.properties[DBUS_ATTR_DEVICE]).decode()
+
+    @property
+    @dbus_property
+    def id(self) -> str:
+        """Return unique identifer."""
+        return self.properties[DBUS_ATTR_ID]
+
+    @property
+    @dbus_property
+    def read_only(self) -> bool:
+        """Return whether device is read only."""
+        return self.properties[DBUS_ATTR_READ_ONLY]
+
+    @property
+    @dbus_property
+    def symlinks(self) -> list[str]:
+        """Return list of symlinks."""
+        return [
+            bytes(symlink).decode() for symlink in self.properties[DBUS_ATTR_SYMLINKS]
+        ]
+
+    @property
+    @dbus_property
+    def device_number(self) -> int:
+        """Return device number."""
+        return self.properties[DBUS_ATTR_DEVICE_NUMBER]
+
+    @property
+    @dbus_property
+    def configuration(
+        self,
+    ) -> dict[str, list[FstabConfigDetails | CrypttabConfigDetails]]:
+        """Return device configuration."""
+        configuration = defaultdict(list)
+        for type_, details in self.properties[DBUS_ATTR_CONFIGURATION]:
```

Comment:
```suggestion
        for type_, details in self.properties[DBUS_ATTR_CONFIGURATION].items():
```

Reply:
Oh wait this is my bad. This isn't a dictionary. Undoing this, sorry about that.